src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.EcDsa.cs

```diff
@@ -25,5 +28,49 @@ internal static partial class Crypto
         // returns the maximum length of a DER encoded ECDSA signature created with this key.
         [DllImport(Libraries.CryptoNative, EntryPoint = "CryptoNative_EcDsaSize")]
         internal static extern int EcDsaSize(SafeEcKeyHandle ecKey);
+
+        // todo: move this to common cng and openssl
+        [DllImport(Libraries.CryptoNative, EntryPoint = "CryptoNative_LookupNidByOidValue")]
+        private static extern int CryptoNative_LookupNidByOidValue(string oidValue, ref int nid);
+
+        [DllImport(Libraries.CryptoNative, EntryPoint = "CryptoNative_LookupNidByFriendlyName")]
+        private static extern int CryptoNative_LookupNidByFriendlyName(string friendlyName, ref int nid);
+        
+        internal static int OidToNid(Oid oid)
+        {
+            if (oid == null)
+            {
+                throw new ArgumentNullException(nameof(oid));
+            }
+
+            if (string.IsNullOrEmpty(oid.Value) && string.IsNullOrEmpty(oid.FriendlyName))
+            {
+                throw new ArgumentException(string.Empty, nameof(oid));
+            }
+
+            int nid = Interop.Crypto.NID_undef;
+            int result = 0;
+            
+            if (!string.IsNullOrEmpty(oid.Value))
+                result = Interop.Crypto.CryptoNative_LookupNidByOidValue(oid.Value, ref nid);
+
+            if (result == 0 && !string.IsNullOrEmpty(oid.FriendlyName))
+                result = Interop.Crypto.CryptoNative_LookupNidByFriendlyName(oid.FriendlyName, ref nid);
```

Comment:
Yes (curve25519?) and a handful of others without an oid value. Also note (by design the way I have it currently) that nothing currently prevents someone from doing the same by passing in a hand-crafted oid to ECNamedCurve() ctor that only contains the friendly name -- i.e. we don't force the use of the dotted value.